lib/matplotlib/colorbar.py

```diff
@@ -25,7 +25,7 @@
 import matplotlib.scale as mscale
 import matplotlib.spines as mspines
 import matplotlib.transforms as mtransforms
-from matplotlib import docstring
+from matplotlib import _docstring
```

Comment:
Merge with line 21?

Reply:
Sure. These things are not so obvious, as it seems a bit arbitrary to me how it is handled in different files (like the original version of this file...).

Reply:
These imports have evolved historically. Nobody has taken the effort to enforce a consinstent set of rules. Generally I'd say:

- group with empty lines in between the blocks
  - standard library
  - other libraries, e.g. numpy
  - matplotlib imports
- Use `from matpotlib import _api, cbook, ...` in case of multiple submodules. Use alphabetic sorting, including all `_` first.
- Keep aliased imports on a separate line as `import matplotlib.scale as mscale`.  
  Don't use the form `from matplotlib import scale as mscale`, and in particular not `from matplotlib import scale as mscale, spines as mspines`. That's not readable.
  At some point we should decise whether we want such aliasing, but leave them as is for now.

This is not a full specification and is not consistently applied in the codebase. But that's what I'd do for new code.